podtato-head-monolith/podtatoserver.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/podtato-head/podtato-head-monolith/pkg"
+
+	"github.com/gorilla/mux"
+
+	"html/template"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var podtatoConfiguration *pkg.PodtatoConfig
+
+// HTML page template
+var overviewTemplate *template.Template
+
+// create a new counter vector
+var getCallCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "http_requests_total", // metric name
+		Help: "Number of get requests.",
+	},
+	[]string{"status"}, // labels
+)
+
+var buckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}
+
+var responseTimeHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "http_server_request_duration_seconds",
+	Help:    "Histogram of response time for handler in seconds",
+	Buckets: buckets,
+}, []string{"route", "method", "status_code"})
+
+// create a handler struct
+type HTTPHandler struct{}
+
+type statusRecorder struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (rec *statusRecorder) WriteHeader(statusCode int) {
+	rec.statusCode = statusCode
+	rec.ResponseWriter.WriteHeader(statusCode)
+}
+
+func getRoutePattern(r *http.Request) string {
+	reqContext := mux.CurrentRoute(r)
+	if pattern, _ := reqContext.GetPathTemplate(); pattern != "" {
+		return pattern
+	}
+
+	fmt.Println(reqContext.GetPathRegexp())
+
+	return "undefined"
+}
+
+// implement `ServeHTTP` method on `HttpHandler` struct
+func (h HTTPHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	var status string
+	defer func() {
+		// increment the counter on defer func
+		getCallCounter.WithLabelValues(status).Inc()
+	}()
+
+	overviewTemplate = template.Must(template.ParseFiles("./static/podtato-new.html"))
+	err := overviewTemplate.Execute(res, podtatoConfiguration)
+
+	if err != nil {
+		log.Print(err.Error())
+	}
+	// Slow build
+	if podtatoConfiguration.ServiceVersion == "0.1.2" {
+		time.Sleep(2 * time.Second)
```

Comment:
Maybe change this delay to something smaller e.g. 400 milliseconds to reduce waiting time for tests with 1000+ requests.

Reply:
A reasonable idea, but I think it should be in a follow-on PR so we can keep things here as close as possible to how they were originally.

Reply:
Yes that sounds reasonable.